cesium/internal/domain/index.go

```diff
@@ -102,45 +163,11 @@ func (idx *index) beforeFirst(ts telem.TimeStamp) bool {
 	return ts.Before(idx.mu.pointers[0].Start)
 }
 
-func (idx *index) insertAt(ctx context.Context, i int, p pointer) {
-	idx.modifyAfter(ctx, i, func() {
-		if i == 0 {
-			idx.mu.pointers = append([]pointer{p}, idx.mu.pointers...)
-		} else if i == len(idx.mu.pointers) {
-			idx.mu.pointers = append(idx.mu.pointers, p)
-		} else {
-			idx.mu.pointers = append(idx.mu.pointers[:i], append([]pointer{p}, idx.mu.pointers[i:]...)...)
-		}
-	})
-}
-
-func (idx *index) updateAt(ctx context.Context, i int, p pointer) (err error) {
-	ptrs := idx.mu.pointers
-	idx.modifyAfter(ctx, i, func() {
-		oldP := ptrs[i]
-		if oldP.Start != p.Start {
-			err = RangeNotFound
-			return
-		}
-		overlapsWithNext := i != len(ptrs)-1 && ptrs[i+1].OverlapsWith(p.TimeRange)
-		overlapsWithPrev := i != 0 && ptrs[i-1].OverlapsWith(p.TimeRange)
-		if overlapsWithPrev || overlapsWithNext {
-			err = ErrDomainOverlap
-		} else {
-			idx.mu.pointers[i] = p
-		}
-	})
-	return
-}
-
-func (idx *index) modifyAfter(ctx context.Context, i int, f func()) {
-	update := indexUpdate{afterIndex: i}
-	defer func() {
-		idx.mu.Unlock()
-		idx.Observer.Notify(ctx, update)
-	}()
-	idx.mu.Lock()
-	f()
+func (idx *index) persist(ctx context.Context) {
```

Comment:
Why aren't you calling this function on line 78 instead of calling Notify explicitly?